src/main/java/com/google/codeu/data/Message.java

```diff
@@ -28,6 +28,7 @@
   private String text;
   private String subject;
   private long timestamp;
+  private boolean isPublic;
```

Comment:
You don't need this variable and can just `return user.equals(recipient)` from the getPublic() method--since this class doesn't change, it should be the same either way, so not a big difference.

Reply:
done